core/headerchain.go

```diff
@@ -122,87 +122,121 @@ func (hc *HeaderChain) GetBlockNumber(hash common.Hash) *uint64 {
 	return number
 }
 
-// WriteHeader writes a header into the local chain, given that its parent is
-// already known. If the total difficulty of the newly inserted header becomes
-// greater than the current known TD, the canonical chain is re-routed.
+// WriteHeaders writes a chain of headers into the local chain, given that the parents
+// are already known. If the total difficulty of the newly inserted chain becomes
```

Comment:
"are already known or included in the chain of headers being passed in"?

Reply:
Both, the first header needs to have it's parent already known, and the rest of the headers must be a contiguous chain